app/scripts/controllers/certificates.js

```diff
@@ -201,6 +203,37 @@ angular.module('openhimConsoleApp')
       }
     };
 
+    $scope.addPassphrase = function () {
+      Alerting.AlertReset();
+      $scope.certificateObject = new Api.Keystore();
+      $scope.certificateObject.passphrase = $scope.serverPassphrase;
+      $scope.certificateObject.$save({ type: 'passphrase' }, function(){
+        $scope.passphraseSuccess('serverKey', '');
+      }, function(err){
+        $scope.passphraseFail(err, 'serverKey', '');
+      });
+    };
+
+    $scope.passphraseSuccess = function(location){
+      Api.Keystore.get({ type: 'validity' }, function(result){
+        $scope.certValidity = result;
+        Alerting.AlertAddMsg(location, 'success', 'Passphrase submitted');
```

Comment:
Maybe instead of saying 'passphrase submitted' here we should say 'Passphrase matches supplied key'. This will work better with the fail case.